torch/_torch_docs.py

```diff
@@ -13669,6 +13669,36 @@ def merge_dicts(*dicts):
 """,
 )
 
+add_docstr(
+    torch.register_privateuse1_backend,
```

Comment:
Not sure about the name here.
In particular, keeping in the back of my mind that we might need to extend this to support multiple backends.

I am wondering if we want to make an API that will support the multi backend case directly here. And thus naming it something like `torch.register_new_backend("foo")`. And then from c++ recommend they use `getDispatchKeyForDeviceType(parse_type("foo"))` (not sure what is the exact name for this?) to register their kernels.

The other option is to say they need to use PrivateUse1 manually and thus this API is just to rename it and much more limited. In such a case, I would suggest we move it to a smaller namespace like `torch.utils.rename_privateuse1_backend("foo")`.

Neither of these options require changing the way things are implemented right now btw, just the python naming, the doc and the error message when multiple backends are being used.
WDYT?

Reply:
> And then from c++ recommend they use `getDispatchKeyForDeviceType(parse_type("foo"))`

I don't think this will easily work without some bigger API changes, since right now on the C++ side you still need to hardcode the PrivateUse1 key name directly into our registration macro: `TORCH_LIBRARY_IMPL(aten, PrivateUse1, m) {...}`.

If we wanted to fully go down the multi-backend route now (and abstract PrivateUse1 away from any public API's), one way would be with a new macro like `TORCH_BACKEND_IMPL(aten, foo, m {...})` where we set up the link between a PrivateUse key and the backend name at library load time. That would make a python API like `torch.register_new_backend()` redundant though.

LMK if you agree with that line of reasoning - but I'm leaning towards your second proposal, which is to use a smaller namespace (`torch.utils.rename_privateuse1_backend`)

Reply:
Sounds good!